docs/data-sources/dli_spark_templates.md

```diff
@@ -0,0 +1,144 @@
+
+  The [dependent_packages](#body_dependent_packages_struct) structure is documented below.
+
+* `app_parameters` - The input parameters of the main class, that is application parameters.
+
+* `obs_bucket` - The OBS bucket for storing the spark jobs.
+
+* `auto_recovery` - Indicates whether to enable the retry function.
+
+* `max_retry_times` - The maximum number of retries.
+
+* `jars` - The name of the resource package of type jar upload to the DLI resource management system.
+
+* `configurations` - The configuration items of the DLI spark.
+
+* `executor_memory` - The executor memory of the spark application.
+
+* `queue_name` - The DLI queue name.
+
+* `main_class` - The spark main class of the template.
+
+* `python_files` - Name of the package that is of the PyFile type.
+  And has been uploaded to the DLI resource management system.
+
+* `files` - The name of the resource package of type file upload to the DLI resource management system.
+
+* `modules` - The name of the dependent system resource module.
+  DLI provides dependencies for executing datasource jobs.
+  The dependent modules and corresponding services are as follows.
+  + **sys.datasource.hbase**: CloudTable/MRS HBase
+  + **sys.datasource.opentsdb**: CloudTable/MRS OpenTSDB
+  + **sys.datasource.rds**: RDS MySQL
+  + **sys.datasource.css**: CSS
+
+* `driver_memory` - The driver memory of the spark application.
+
+* `executor_cores` - The number of CPU cores of each executor in the spark application.
+
+* `app_name` - The name of the uploaded JAR or pyFile type package.
+
+* `specification` - The compute resource type. Currently, resource types A, B, and C are available.  
+  The available types and related specifications are as follows, default to minimum configuration (type **A**).
+  | type | resource | driver cores | executor cores | driver memory | executor memory | num executor |
+  | ---- | ---- | ---- | ---- | ---- | ---- | ---- |
+  | A | 8 vCPUs, 32-GB memory | 2 | 1 | 7G | 4G | 6 |
+  | B | 16 vCPUs, 64-GB memory | 2 | 2 | 7G | 8G | 7 |
+  | C | 32 vCPUs, 128-GB memory | 4 | 2 | 12G | 8G | 14 |
+
+* `resources` - The list of resource objects.
+
+  The [resources](#body_resources_struct) structure is documented below.
+
+* `driver_cores` - The number of CPU cores of the spark application driver.
+
+* `num_executors` - The number of executors in a spark application.
+
+<a name="body_dependent_packages_struct"></a>
+The `dependent_packages` block supports:
+
+* `name` - The name of a user group.
+
+* `resources` - The resources of a user group.
+
+  The [resources](#dependent_packages_resources_struct) structure is documented below.
+
+<a name="dependent_packages_resources_struct"></a>
+The `resources` block supports:
+
+* `name` - The name of resource.
+
+* `type` - The type of resource.
+
+<a name="body_resources_struct"></a>
+The `resources` block supports:
+
+* `name` - The name of resource.
+
+* `type` - The type of resource.
```

Comment:
The same structure definition does not need to be declared twice.
The same H5 tag can be referenced.

Reply:
done